docs/documentation/platform/kms/encryption-strategies.mdx

```diff
@@ -0,0 +1,171 @@
+
+                You need to follow the instructions provided by the HSM provider to set up the HSM device. Once the HSM device is set up, the HSM device can be used within Infisical.
+
+                After setting up the HSM from your provider, you will be given a PKCS#11 library that will be used to interface with the HSM. This library should be placed in a secure location on the same machine as the Infisical instance(s).
+            </Step>
+            <Step title="Configure HSM on Infisical">
+                Configuring the HSM on Infisical requires setting a set of environment variables.
+                - `HSM_LIB_PATH`: The path to the PKCS#11 library provided by the HSM provider. This usually comes in the form of a `.so` for Linux and MacOS, or a `.dll` file for Windows. For Docker, you need to mount the library path as a volume. Further instructions can be found below.
+                - `HSM_PIN`: The PKCS#11 PIN to use for authentication with the HSM device.
+                - `HSM_SLOT`: The slot number to use for the HSM device. This is usually `0` for most HSM devices.
+                - `HSM_KEY_LABEL`: The label of the key to use for encryption. **Please note that if no key is found with the provided label, the HSM will create a new key with the provided label.**
+
+                You can read more about the [default instance configurations](/self-hosting/configuration/envars) here.
+            </Step>
+
+            <Step title="Restart Instance">
+                After setting up the HSM, you need to restart the Infisical instance for the changes to take effect.
+            </Step>
+
+            <Step title="Set encryption strategy to HSM">
+                After restarting the Infisical instance, you need to navigate to your Infisical Console, and head into the Server Admin Console. In the Server Admin Console, open the Encryption section, and set the KMS encryption strategy to HSM.
+
+                <Steps>
+                    <Step title="Navigate to admin console">
+                        ![Server Admin Console](/images/platform/kms/hsm/server-admin-console.png)
+                    </Step>
+                    <Step title="Update encryption strategy">
+                        ![Set Encryption Strategy](/images/platform/kms/hsm/encryption-strategy.png)
+
+                        Once you press the 'Save' button, your Infisical instance will immediately switch to the HSM encryption strategy. This will re-encrypt your KMS key with keys from the HSM device.
+                    </Step>
+                </Steps>
```

Comment:
I would suggest not nesting these steps and just breaking them out and including them in the root steps separately or just under one step